core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -175,8 +175,7 @@ class Replica(val brokerId: Int,
   def convertHWToLocalOffsetMetadata() {
     if (isLocal) {
       highWatermarkMetadata = log.get.convertToOffsetMetadata(highWatermarkMetadata.messageOffset).getOrElse {
-        val firstOffset = log.get.logSegments.head.baseOffset
-        new LogOffsetMetadata(firstOffset, firstOffset, 0)
+        new LogOffsetMetadata(logStartOffset, logStartOffset, 0)
```

Comment:
Are we guaranteed that the log start offset is also the base offset of the segment it is in? I am not too sure if it matters or not, but the reason I ask is because of `LogOffsetMetadata.onSameSegment`, which raises an exception in a fetch request if the base offset is not correct.

Reply:
@hachikuji Thanks much for catching this. Sorry, I made this patch too quick without checking reviewing it myself. I have updated the patch to fix the issue.

Reply:
@hachikuji Can you review the current solution?